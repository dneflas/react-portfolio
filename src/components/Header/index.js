import React from "react";
import Nav from "../Navigation";

function Header({ currentSection, setCurrentSection }) {
  return (
    <header>
      <div className="container flex-row justify-space-between align-center pt-3">
        <h1 className="page-title py-2 px-3">Debbie Neflas</h1>
        <Nav
          currentSection={currentSection}
          setCurrentSection={setCurrentSection}
        />
      </div>
    </header>
  );
}

export default Header;
