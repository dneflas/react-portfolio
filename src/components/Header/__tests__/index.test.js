import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "..";

const mockCurrentSection = jest.fn();
const mockSetCurrentSection = jest.fn();

afterEach(cleanup);

describe("Header renders", () => {
  it("renders", () => {
    render(<Header />);
  });
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Header
        currentSection={mockCurrentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("title is visible", () => {
    const { getByTestId } = render(
      <Header
        currentSection={mockCurrentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
    expect(getByTestId("title")).toHaveTextContent("Debbie Neflas");
  });
});
