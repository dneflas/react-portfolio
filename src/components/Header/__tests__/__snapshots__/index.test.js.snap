// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="container flex-row justify-space-between align-center pt-3"
      data-testid="title"
    >
      <h1
        class="page-title py-2 px-3"
      >
        Debbie Neflas
      </h1>
      <nav
        class="flex-row"
      >
        <a
          class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
          data-testid="about"
          href="#about"
        >
          About Me
        </a>
        <a
          class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
          data-testid="portfolio"
          href="#portfolio"
        >
          Portfolio
        </a>
        <a
          class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
          data-testid="contact"
          href="#contact"
        >
          Contact
        </a>
        <a
          class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
          data-testid="resume"
          href="#resume"
        >
          Resume
        </a>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
